lib/dialects/oracledb.ts

```diff
@@ -141,17 +122,19 @@ export default class oracleDB implements SchemaInspector {
     const query = this.knex
       .with(
         'uc',
-        this.knex.raw(
-          'SELECT "TABLE_NAME", "CONSTRAINT_NAME", "R_CONSTRAINT_NAME", "CONSTRAINT_TYPE" FROM "USER_CONSTRAINTS"'
-        )
-      )
-      .with(
-        'ucc',
-        this.knex.raw(
-          'SELECT "TABLE_NAME", "COLUMN_NAME", "CONSTRAINT_NAME" FROM "USER_CONS_COLUMNS"'
-        )
+        this.knex.raw(`
+          SELECT /*+ materialize */ DISTINCT
```

Comment:
What in the holy oracle is this black magic

Reply:
It's a forced optimization. It tells Oracle to store the results of the query so they can be accessed faster. This optimization is usually applied anyway, but not always.

Reply:
This can be kinda important. For example, the main SchemaOverview query in Directus takes twice as long without materialize (after I refactored it... PR incoming).

Reply:
Nice! The more you know 🌈 ⭐